### ISSUE TITLE
Deletion errs with `Get "http://localhost...serviceaccounts/k8s-manifests-deployer-service-account...`

### ISSUE THREAD

**Issue by NimJay** [GoogleCloudPlatform/terraform-ecommerce-microservices-on-gke#40]
* This error occurred on Jump Start Solutions (JSS) in the Google Cloud Console.
* When we tried deleting this JSS, we got the following error:
```
time="2023-05-12T00:31:23Z" level=error msg="Error (exit code 1) running \"terraform destroy -auto-approve -json\". Stderr:\n"
Error: Get "http://localhost/api/v1/namespaces/default/serviceaccounts/k8s-manifests-deployer-service-account": dial tcp 127.0.0.1:80: connect: connection refused
running tf destroy: terraform destroy failed: running terraform failed: exit status 1
Error: running tf destroy: terraform destroy failed: running terraform failed: exit status 1
```
* When I `terraform destroy`-ed locally, the resources successfully got destroyed:
```
....
kubernetes_service_account.kubernetes_manifests_deployer_service_account: Destroying... [id=default/k8s-manifests-deployer-service-account]
kubernetes_service_account.kubernetes_manifests_deployer_service_account: Destruction complete after 0s
...
Destroy complete! Resources: 28 destroyed.
```
* Before I `terraform destroyed`-ed locally, I downloaded the JSS's `terraform.tfstate` file from the Google Cloud Storage bucket (along with `kubernetes_provider.tf` and `helm_provider.tf` which are files [re]generated by the Terraform itself).
* The fact that my local `terraform destroy` worked suggests that there was nothing wrong with the `host` or `cluster_ca_certificate` (inside the Terraform-generated `kubernetes_provider.tf` file) [hardcoded by Terraform](https://github.com/GoogleCloudPlatform/terraform-ecommerce-microservices-on-gke/blob/32d66f792aa8c15a4d2cd5d887136837336e27fd/infra/deploy_kubernetes_manifests.tf#L37).

### Additional info
* This was reported by @askmeegs.
* Time of creation: May 11, 2023, 8:03:53 PM
* Time of deletion (erroneous): May 12, 2023, 12:22:43 PM
* Successful `terraform apply` [Cloud Build logs here](https://pantheon.corp.google.com/cloud-build/builds;region=us-central1/2e21c020-8723-4f02-94ba-f5e14bd46f06?project=mokeefe-test-1).
* Failed `terraform destroy` [Cloud Build logs here](https://pantheon.corp.google.com/cloud-build/builds;region=us-central1/143a17ce-39bc-40c0-8171-cf0910426555?project=mokeefe-test-1).

**Comment by kweinmeister** [comment]
@NimJay is this issue still valid?

**Comment by NimJay** [comment]
@kweinmeister, yes, this issue is still valid.
I was able to reproduce it.
I believe it only happens when you try to delete the JSS deployment more than 1 day _after_ deployment.
Did you encounter the same issue? If so, do you remember how long after deployment you tried deleting?

I'm going to raise the priority on this. Hoping to have some more insight into the root cause tomorrow morning.

**Comment by NimJay** [comment]
**An update on Jun 9:**
After reproducing this issue in my own Google Cloud project, I followed the instructions in the solution guide, [Delete a console-based deployment by using Terraform](https://cloud.google.com/architecture/application-development/ecommerce-microservices#delete_a-based_deployment_by_using_terraform).
When I followed those instructions on Cloud Shell, the `terraform destroy` didn't give me the same error (during the state refresh), which is strange.
In other words:
**1.** the `terraform destroy` was now
**2.** for now, users can resort [the solution guide](https://cloud.google.com/architecture/application-development/ecommerce-microservices#delete_a-based_deployment_by_using_terraform) to destroy the solution 

I did notice JSS was using `terraform 1.2.3` while my Cloud Shell was using `terraform 1.4.6`.

**Comment by NimJay** [comment]
* I'm reducing the priority of this (from P1 to P2) since we have the solution guide's instructions as a workaround (see previous comment).
* I have also confirmed that the use of `terraform 1.2.3` (by JSS) is not the issue.

**Comment by NimJay** [comment]
This issue should be resolve by https://github.com/GoogleCloudPlatform/terraform-ecommerce-microservices-on-gke/pull/55.
